### PR TITLE
Fix ruff linting 12

### PR DESCRIPTION
**Pull request by gabe-l-hart**
## Related Issue
Supports #12 

## What this PR does / why we need it

This PR fixes all `ruff` linting errors and enables linting as part of CI and `pre-commit`

## What gif most accurately describes how I feel towards this PR?
Example of a gif

### CODE REVIEW

File: oper8/reconcile.py [hunk 1]
```diff
@@ -79,14 +78,16 @@ class ReconciliationResult:
     exception: Exception = None
 
 
+# Forward declarations of Controller
+CONTROLLER_TYPE = "Controller"
```
Comment (HonakerM) [inline oper8/reconcile.py:82]: Should the forward declarations all be in the same place? Maybe in `consts`?

File: oper8/component.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def __init__(
         """
         # Ensure that the name property is defined by accessing it and that
         # namespace is inherited from session.
-        self.name  # pylint: disable=pointless-statement
+        self.name  # noqa: B018
```
Comment (HonakerM) [inline oper8/component.py:58]: This is very much a NIT/low ca but does ruff support long form disables? I always have issues trying to figure out what the shorthand means.

File: oper8/watch_manager/python_watch_manager/utils/log_handler.py [hunk 3]
```diff
@@ -14,14 +13,17 @@
...
+
 
 class LogQueueHandler(QueueHandler):
     """
     Log Handler class to collect messages from a child processes and pass
     them to the root process via a multiprocess queue
     """
 
-    def __init__(self, queue: "Queue[Any]", manifest: ManagedObject = None):
+    def __init__(self, queue: QUEUE_TYPE, manifest: ManagedObject = None):
```
Comment (HonakerM) [inline oper8/watch_manager/python_watch_manager/utils/log_handler.py:26]: I'm not sure why I didn't reference Queue directly here. Is there a circular import somehow with multiprocessing?

**Review by HonakerM — APPROVED** [review]

File: oper8/reconcile.py (hunk 1 above)
Comment (gabe-l-hart) [reply oper8/reconcile.py:82]: I thought about that. I was trying to follow the `c++` paradigm of putting them in the "header" section of the individual file to handle circular dependencies.

File: oper8/component.py (hunk 2 above)
Comment (gabe-l-hart) [reply oper8/component.py:58]: Ah, yeah, TBH I don't know. I agree that the shorthand is much less readable, but I'm not aware of a long-form alias system

Comment (HonakerM) [reply oper8/component.py:58]: Okay no worries! I saw there's a list of them in `pyproject.toml` so I think that's enough

File: oper8/reconcile.py (hunk 1 above)
Comment (HonakerM) [reply oper8/reconcile.py:82]: I think that makes sense, it duplicates some code but will be a bit clearer in the long run

**Pull request merged**